src/main/java/seedu/duke/Save.java

```diff
@@ -1,37 +1,42 @@
 package seedu.duke;
 
-
 import seedu.duke.ai.Ai;
 import seedu.duke.stats.MatchStat;
 import seedu.duke.ui.Ui;
 
-
 public class Save {
 
-
     public static void executeSave() {
         int userSaveDirection = getUserSaveDirection();
         int aiPenaltyDirection = Ai.getAiDirection();
         boolean isGoalSaved = userSaveDirection == aiPenaltyDirection;
-
-
         Formatter.printSaveResult(isGoalSaved);
         MatchStat.updateStat(isGoalSaved);
     }
 
-
     private static int getUserSaveDirection() {
-        int userDirection;
-        do {
+        int userDirection = -1;
+        while (userDirection == -1) {
             System.out.print("Enter the direction to save the penalty (0, 1, or 2): ");
             String directionString = Ui.IN.nextLine().trim();
-            userDirection = ShootDirectionConverter.convertToShootDirection(directionString);
+            userDirection = convertToShootDirection(directionString);
             if (userDirection == -1) {
                 System.out.println("Invalid direction! Please enter 0, 1, or 2.");
             }
-        } while (userDirection == -1);
+        }
         return userDirection;
     }
 
-
+    private static int convertToShootDirection(String direction) {
+        try {
+            int directionIndex = Integer.parseInt(direction);
```

Comment:
Can you parse it to int? Because you are trying to parse "save int"

Reply:
The convertToShootDirection() method is designed to parse the argument for the "save" command, which should be an integer within the range (0, 1, or 2). The string passed to this method is expected to be just the argument part of the input, not the full command. In the case of the "save 1" command, only "1" would be passed to convertToShootDirection(). This is handled by the Parser class, which splits the full input string into the command and argument tokens.